src/core/kdtree.hpp

```diff
@@ -240,8 +241,9 @@ class Kdtree {
 		NOT_REACHED(); // a.first == b.first: same element must not be inserted twice
 	}
 	/** Search a sub-tree for the element nearest to a given point */
-	node_distance FindNearestRecursive(CoordT xy[2], size_t node_idx, int level) const
+	node_distance FindNearestRecursive(CoordT xy[2], size_t node_idx, int level, DistT limit = (DistT)(MapSizeX() + MapSizeY())) const
```

Comment:
I don't really like this having a non-constant default value for a parameter.
Could you change it to use `std::numeric_limits<DistT>::max()` instead?